src/taipy/config/_config_comparator/_config_comparator.py

```diff
@@ -0,0 +1,156 @@
+
+
+class _ConfigComparator:
+
+    _UNCONFLICTED_SECTIONS: Set[str] = set()
+
+    __logger = _TaipyLogger._get_logger()
+
+    @classmethod
+    def _add_unconflicted_section(cls, section_name: Union[str, Set[str]]):
+        if isinstance(section_name, str):
+            section_name = {section_name}
+
+        cls._UNCONFLICTED_SECTIONS.update(section_name)
+
+    @classmethod
+    def _compare(
+        cls,
+        old_config: _Config,
+        new_config: _Config,
+        old_version_number: Optional[str] = None,
+        new_version_number: Optional[str] = None,
+        raise_error: bool = True,
+    ):
+        """Compare between 2 _Config object to check for compatibility.
+
+        Args:
+            old_config (_Config): The old _Config.
+            new_config (_Config): The new _Config.
+            old_version_number (str, optional): The old version number for logging. Defaults to None.
+            new_version_number (str, optional): The new version number for logging. Defaults to None.
+            raise_error (bool, optional): Raise exception if there is error or not. Defaults to True.
+
+        Raises:
+            ConflictedConfigurationError: Raise if there is conflicted between the 2 _Config and raise_error is True.
+
+        Returns:
+            _ComparatorResult: Return a _ComparatorResult dictionary with the following format:
+        ```python
+        {
+            "added_items": [
+                ((section_name_1, config_id_1, attribute_1), added_object_1),
+                ((section_name_2, config_id_2, attribute_2), added_object_2),
+            ],
+            "removed_items": [
+                ((section_name_1, config_id_1, attribute_1), removed_object_1),
+                ((section_name_2, config_id_2, attribute_2), removed_object_2),
+            ],
+            "modified_items": [
+                ((section_name_1, config_id_1, attribute_1), (old_value_1, new_value_1)),
+                ((section_name_2, config_id_2, attribute_2), (old_value_2, new_value_2)),
+            ],
+        }
+        ```
+        """
+        old_json_config = json.loads(_JsonSerializer._serialize(old_config))
+        new_json_config = json.loads(_JsonSerializer._serialize(new_config))
+
+        config_deepdiff = DeepDiff(old_json_config, new_json_config)
+
+        comparator_result = _ComparatorResult(copy(cls._UNCONFLICTED_SECTIONS))
+
+        comparator_result._check_added_items(config_deepdiff, new_json_config)
+        comparator_result._check_removed_items(config_deepdiff, old_json_config)
+        comparator_result._check_modified_items(config_deepdiff, old_json_config, new_json_config)
+        comparator_result._sort_by_section()
+
+        cls.__log_message(comparator_result, old_version_number, new_version_number)
+
+        if raise_error:
+            if comparator_result.get(_ComparatorResult.CONFLICTED_SECTION_KEY):
+                raise ConflictedConfigurationError
```

Comment:
This should be removed from here. It should be added to the client side (VersionManager, I guess)